cloud/aws/services/ec2/instances.go

```diff
@@ -39,47 +39,48 @@ func (s *Service) InstanceIfExists(instanceID *string) (*v1alpha1.Instance, erro
 	}
 
 	if len(out.Reservations) > 0 && len(out.Reservations[0].Instances) > 0 {
```

Comment:
should this throw error cases if those lens are zero? it's out of scope for this PR, but maybe an improvement for later

Reply:
hmm, maybe a not found error?